Source/Decode.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  Decodable.swift
+//
+//  Copyright (c) 2015-2016 Nike, Inc. (https://www.nike.com)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+
+// MARK: Decodable Parsing Methods
+
+/**
+ Parses json data at the specified key path into an object of type `T`. `T` must implement the `Decodable` protocol.
+
+ - parameter data:       A Data object containing encoded json data.
+ - parameter forKeyPath: The json key path identifying the object to be parsed. Default is `""`.
+
+ - throws:  A ParserError.Deserialization and ParserError.Validation error if parsing fails.
+ - returns: The parsed object.
+ */
```

Comment:
We need to port Elevate over to using the new formatting conventions. Would be as follows:

```swift
/// Parses json data at the specified key path into an object of type `T`. `T` must implement the `Decodable` protocol.
/// 
/// - parameter data:       A Data object containing encoded json data.
/// - parameter forKeyPath: The json key path identifying the object to be parsed. Default is `""`.
/// 
/// - throws:  A ParserError.Deserialization and ParserError.Validation error if parsing fails.
/// - returns: The parsed object.
```

Wouldn't take long to update the entire project since there aren't too many public APIs.